src/components/Rocket.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { bookRockets, cancelRocketBooking } from '../redux/rockets';
+/* eslint-disable react/prop-types */
+const Rocket = (props) => {
+  const { rocket } = props;
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    if (rocket.reserved && rocket.reserved === true) {
+      dispatch(cancelRocketBooking(rocket.id));
+    } else {
+      dispatch(bookRockets(rocket.id));
+    }
+  };
```

Comment:
"Reserve Rocket" button updates the store successfully by dispatching an action 🤯 Well done guys ✔